amp/WEB-INF/src/org/digijava/module/aim/util/CurrencyUtil.java

```diff
@@ -765,25 +770,40 @@ public static void checkDatabaseSanity(Session session) throws AimException {
                     + "3) from_currency_code <> to_currency_code; " + System.lineSeparator()
                     + "4) unique tuples (from, to, date).";
         }
-        if ( errMsg!=null ) {
+        if (errMsg != null) {
             logger.error("AMP_CURRENCY_RATE table consistency check - FAIL:" + errMsg);
             throw new AimException(errMsg);
         } else {
             logger.debug("AMP_CURRENCY_RATE table consistency check - PASS");
         }
     }
-    
+
     public static void deleteCurrencyRates(String currencyCode) {
         PersistenceManager.getSession().createQuery(
-                String.format("delete from %s o where o.fromCurrencyCode = '%s' or o.toCurrencyCode = '%s'", 
+                String.format("delete from %s o where o.fromCurrencyCode = '%s' or o.toCurrencyCode = '%s'",
                         AmpCurrencyRate.class.getName(), currencyCode, currencyCode)).executeUpdate();
     }
-    
+
     public static void deleteCurrencyRates(List<String> currencyCode) {
         if (currencyCode.size() == 0)
             return;
         String codes = Util.toCSString(currencyCode);
-        PersistenceManager.getSession().createSQLQuery(String.format("DELETE FROM amp_currency_rate r " 
+        PersistenceManager.getSession().createSQLQuery(String.format("DELETE FROM amp_currency_rate r "
                 + " WHERE r.from_currency_code in (%s) OR r.to_currency_code in (%s)", codes, codes)).executeUpdate();
     }
+
+    public static AmpCurrency getEffectiveCurrency() {
+
+        AmpCurrency curr = null;
+        TeamMember tm = (TeamMember) TLSUtils.getRequest().getSession().getAttribute("currentMember");
+        if (tm != null && tm.getAppSettings() != null && tm.getAppSettings().getCurrencyId() != null) {
+            curr = CurrencyUtil.getAmpcurrency(tm.getAppSettings().getCurrencyId());
+        } else {
+            String currCode = FeaturesUtil.getGlobalSettingValue(GlobalSettingsConstants.BASE_CURRENCY);
+            if (currCode != null) {
+                curr = CurrencyUtil.getAmpcurrency(currCode);
+            }
+        }
+        return curr;
+    }
```

Comment:
This can be simplified to one line `CurrencyUtil.getWorkspaceCurrency(TeamUtil.getCurrentMember())`